packages/modules/src/modules/epics/ContributionsEpic.tsx

```diff
@@ -253,6 +261,72 @@ const EpicBody: React.FC<BodyProps> = ({
     );
 };
 
+/*
+Issues I'm having:
+
+    - This code properly belongs as a function in server/src/lib/ab.ts 
+    but I've struggled too long with types to get it working there
+
+    - Messaging test variants can suppress the Choice Cards. Need to check that 
+    the new code accounts for this and doesn't record users as being part of
+    an amounts test if they see a contribs epic with no choice card displayed?
+
+    - This code reflects (as far as possible) what happens in support frontend - 
+    as amounts testing can also start on the contributions page there if 
+    a user navigates to the page by means other than the Epic CTA button
+
+    - Can I generate the mvtId number locally here? Currently it's generated 
```

Comment:
the mvt cookie is for the `theguardian.com` domain, so is shared between dotcom + support. So we should use the same mvt value (from the targeting data)

Reply:
This worries me. I thought marketing AB tests and amounts AB tests needed to be completely separate. Using one MVT value will lead to the situation where, assuming both an Epic messaging AB test and an amounts AB test are running, the user will be assigned to messaging and amounts variants based on the same MVT value - so a given message variant will always display a particular amounts variant. This pollutes the amounts test as the decision by the user on how much to contribute can be determined as much by the messaging as by the displayed amounts. Does that make sense?

Reply:
We use the same mvt value everywhere.
Each test has a seed value (either explicitly, or based on the test name), to ensure independent tests do not interfere.
The amounts test data actually includes a seed value, so the SDC backend should use this when assigning the user to an amounts test (exactly the same as on support-frontend)